File: Programming Basics with JS/T08 - Exam Preparation/lecture/06. Christmas Tournament.js
function christmas(input) {
    let days = Number(input[0]);
    let index = 1;
    let command = input[index];
    let totalMoney = 0;
    let daysWon = 0;
    let daysLost = 0;

    for (let i = 0; i < days; i++) {
        let won = 0;
        let lost = 0;
        let tempMoney = 0;

        while (command !== 'Finish') {
            index++;
            let game = input[index];

            if (game === 'win') {
                tempMoney += 20;
                won++;
            } else {
                lost++;
            }
            index++;
            command = input[index];
        }

        if (won > lost) {
            tempMoney = tempMoney + (tempMoney * 10 / 100);
            daysWon++;
        } else {
            daysLost++;
        }
        totalMoney += tempMoney;
        index++;
        command = input[index];
    }

    if (daysWon > daysLost) {
        totalMoney = totalMoney + (totalMoney * 20 / 100);
        console.log(`You won the tournament! Total raised money: ${totalMoney.toFixed(2)}`);
    } else {
        console.log(`You lost the tournament! Total raised money: ${totalMoney.toFixed(2)}`);
    }
}
christmas(["3",
    "darts",
    "lose",
    "handball",
    "lose",
    "judo",
    "win",
    "Finish",
    "snooker",
    "lose",
    "swimming",
    "lose",
    "squash",
    "lose",
    "table tennis",
    "win",
    "Finish",
    "volleyball",
    "win",
    "basketball",
    "win",
    "Finish"]);

File: Programming Basics with Python/T08 - Exam Preparation/lecture/06. Christmas Tournament.py
days = int(input())
money_won = 0
total_wins = 0
total_loses = 0

for day in range(1, days+1):
    result = input()
    money_won_per_day = 0
    wins = 0
    loses = 0

    while result != "Finish":
        result = input()
        if result == "win":
            wins += 1
            money_won_per_day += 20
        elif result == "lose":
            loses += 1
    if wins > loses:
        money_won_per_day *= 1.1
        total_wins += 1
    else:
        total_loses += 1
    money_won += money_won_per_day
total_money = 0

if total_wins > total_loses:
    total_money = money_won * 1.2
    print(f"You won the tournament! Total raised money: {total_money:.2f}")
else:
    total_money = money_won
    print(f"You lost the tournament! Total raised money: {total_money:.2f}")